src/main/java/server/ChatServer.java

```diff
@@ -142,82 +99,113 @@ public static void onlineMessage() throws IOException {
     Socket s;
     boolean isloggedin;
 
+
+    /**
+     * Try to accept the connection. Returns false if the connection is rejected. for any reason.
+     * @param userService
+     * @return
+     */
+    static ClientHandler tryAccept(Socket s, UserService userService) {
+        DataInputStream dis = new DataInputStream(s.getInputStream());
+        PrintWriter dos = new PrintWriter(s.getOutputStream(),true);
+        StringTokenizer st = new StringTokenizer(dis.readLine(), "#");
+        String cmd = st.nextToken();
+
+        try {
+            if (cmd.equals("CONNECT")) {
+                String username = st.nextToken();
+                if (!userService.usernameExists(username)) {
```

Comment:
Note that instead of querying userService for a list of users, we ask the user service if a user exists. This again helps us abstract away things that could potentially be dangerous in a multithreaded environment